apps/src/blockly/addons/cdoFieldDropdown.js

```diff
@@ -32,6 +32,30 @@ export default class CdoFieldDropdown extends GoogleBlockly.FieldDropdown {
     }
   }
 
+  /**
+   * Sets the field's state based on the given state value. Should only be
+   * called by the serialization system.
+   *
+   * @param state The state we want to apply to the field.
+   * @override because different labs store `state` in different ways.
+   * For music lab, `state` is the value of the field.
+   * For other labs, `state` is stringified xml.
+   */
+  loadState(state) {
+    // Check if state is not stringified xml, i.e., value from json.
+    const fieldTagRegEx = /<field/;
+    if (!fieldTagRegEx.test(state)) {
+      if (this.isOptionListDynamic()) {
+        this.getOptions(false);
+      }
+      // TODO: handle config if stored in json
```

Comment:
At this point, when `state` is stored as json. it is assigned only the value. At some point, we may choose support `config` in json so leaving comment here.